src/choleskyQR.jl

```diff
@@ -0,0 +1,170 @@
+"""
+cholesky_jacobimatrix(w, P)
+
+returns the Jacobi matrix `X` associated to a quasi-matrix of polynomials
+orthogonal with respect to `w(x) w_p(x)` where `w_p(x)` is the weight of the polynomials in `P`.
+
+The resulting polynomials are orthonormal on the same domain as `P`. The supplied `P` must be normalized. Accepted inputs are `w` as a function or `W` as an infinite matrix representing multiplication with the function `w` on the basis `P`.
+
+An optional bool can be supplied, i.e. `cholesky_jacobimatrix(sqrtw, P, false)` to disable checks of symmetry for the weight multiplication matrix and orthonormality for the basis (use with caution).
+"""
+function cholesky_jacobimatrix(w::Function, P::OrthogonalPolynomial, checks::Bool = true)
+    checks && !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    W = Symmetric(P \ (w.(axes(P,1)) .* P)) # Compute weight multiplication via Clenshaw
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+function cholesky_jacobimatrix(W::AbstractMatrix, P::OrthogonalPolynomial, checks::Bool = true)
+    checks && !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    checks && !(W isa Symmetric) && error("Weight modification matrix must be symmetric.")
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+
+# The generated Jacobi operators are symmetric tridiagonal, so we store their data as two bands
+mutable struct CholeskyJacobiBands{T} <: AbstractCachedMatrix{T}
+    data::Matrix{T}         # store so-far computed block of entries 
+    U::UpperTriangular      # store upper triangular conversion matrix (needed to extend available entries)
+    X::SymTridiagonal{T}    # store the Jacobi matrix of the original basis (needed to extend available entries)
+    datasize::Int           # size of so-far computed block 
+end
+
+# Computes the initial data for the Jacobi operator bands
+function CholeskyJacobiBands(W::Symmetric{T}, P::OrthogonalPolynomial) where T
+    U = cholesky(W).U
+    X = jacobimatrix(P)
+    dat = zeros(T,2,10)
+    UX = U*X
+    dat[1,1] = (UX * (U \ [1; zeros(∞)]))[1]
+    dat[2,1] = (UX * (U \ [zeros(1); 1; zeros(∞)]))[1]
+    @inbounds for k in 2:10
+        yk = view(UX,k,k-1:k+1)
+        dat[1,k] = dot(yk[1:2], U[k-1:k,k-1:k] \ [0; 1])
```

Comment:
I'm worried about all the allocations, can you do a profile?

Reply:
yup, will take a look

Reply:
ok I think this is improved now, execution time seems fast to me and allocations scale reasonably with block size I think. but if you still see issues or numbers you wouldn't expect let me know